java/cdk/src/main/java/sleeper/cdk/stack/CompactionStack.java

```diff
@@ -356,29 +404,48 @@ private Cluster ecsClusterForCompactionTasks(IBucket configBucket,
                 .build();
         instanceProperties.set(COMPACTION_CLUSTER, cluster.getClusterName());
 
-        FargateTaskDefinition taskDefinition = compactionFargateTaskDefinition("Merge");
-        this.compactionFamily = taskDefinition.getFamily();
-        instanceProperties.set(COMPACTION_TASK_DEFINITION_FAMILY, compactionFamily);
+        FargateTaskDefinition fargateTaskDefinition = compactionFargateTaskDefinition("Merge");
+        String fargateTaskDefinitionFamily = fargateTaskDefinition.getFamily();
+        instanceProperties.set(COMPACTION_TASK_FARGATE_DEFINITION_FAMILY, fargateTaskDefinitionFamily);
+
+        Ec2TaskDefinition ec2TaskDefinition = compactionEC2TaskDefinition("Merge");
+        String ec2TaskDefinitionFamily = ec2TaskDefinition.getFamily();
+        instanceProperties.set(COMPACTION_TASK_EC2_DEFINITION_FAMILY, ec2TaskDefinitionFamily);
 
```

Comment:
Can we only create the task definition, auto-scaling group, etc, that's needed for the deployment mode chosen? e.g. if Fargate is chosen then don't define the Ec2TaskDefinition.

Reply:
Done